src-java/kilda-model/src/main/java/org/openkilda/model/CookieBase.java

```diff
@@ -0,0 +1,162 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.model;
+
+import org.openkilda.exception.InvalidCookieException;
+import org.openkilda.model.bitops.BitField;
+import org.openkilda.model.bitops.NumericEnumField;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@EqualsAndHashCode(of = {"value"})
+public class CookieBase implements Serializable {
```

Comment:
It makes sense to declare the class as abstract. No need to allow instantiating it.

Reply:
done